### DISCUSSION TITLE
Need help with fieldArray where multiple react-select inside a controller

### DISCUSSION THREAD

**Discussion by Adnan0061** [react-hook-form/react-hook-form#8878] [Q&A] [answered]
hello,
I am using a field array for the query builder type option. Now the situation is I have to use 3 **react-select** components where 3rd ones **options** are dependent on the **selected value** of the **first one**. Now If I add all 3 react select inside the same **controller** then conditions can be controlled but as one controller has **one name** I don't get values of all three select. Like in the following image

![1](https://user-images.githubusercontent.com/48943431/185397876-a4879916-2e1c-4508-a6ab-e08fdd1e2a5a.jpg)

But if I use 3 different controllers I can get the data of all 3 **select** but I can't make options conditional according to 1st one's selected options because **field property** is an **internal property** of the **controller**.

![2](https://user-images.githubusercontent.com/48943431/185398473-fe81a92c-e5c5-404c-9152-042b73800d4d.jpg)

how can I solve this actually? the result will be like this.
![image](https://user-images.githubusercontent.com/48943431/185398741-91e595e0-9621-4c7a-b8f7-0bd02b73ffec.png)

**Comment by Moshyfawn** [comment, marked as answer]
You can subscribe to the field value for that field item (row, if you will) via `watch` / `useWatch` and update / maybe reset the dependent field value and options accordingly

**Reply by Adnan0061** [reply]
@Moshyfawn thanks a lot, with **watch** it is easily solved

**Reply by Moshyfawn** [reply]
Awesome!
